lib/factory_bot/strategy/stub.rb

```diff
@@ -44,15 +44,17 @@ def next_id
       end
 
       def stub_database_interaction_on_result(result_instance)
-        result_instance.id ||= next_id
+        if has_settable_id?(result_instance)
+          result_instance.id ||= next_id
```

Comment:
Maybe it's not "doable" but we might want to persist the old behavior. Like assigning the ```primary_key``` the value. That way old versions would stay relevant and even if the primary key is not id it could still be set. we could do:
```ruby
result_instance.send("#{result_instance.class.primary_key}=", next_id)
``` 
Would that cover the case where the primary key is overridden to a field other than id? Or does AR aliases the method to respond to id no matter what?

Reply:
I think `id=` already does the correct thing here for primary keys with a non-standard name. [Here is the relevant rails code](https://github.com/rails/rails/blob/98a57aa5f610bc66af31af409c72173cdeeb3c9e/activerecord/lib/active_record/attribute_methods/primary_key.rb#L22-L25). If we did need to change that behavior, I think it would be beyond the scope of this PR.

Reply:
I tested this out and we shouldn't need any special handling of custom primary keys names in factory_bot.